manager/volume.go

```diff
@@ -311,13 +316,29 @@ func (m *VolumeManager) Attach(name, nodeID string, disableFrontend bool, attach
 		return nil, fmt.Errorf("volume %v is pending restoring", name)
 	}
 
-	// already desired to be attached
+	if v.Spec.NodeID == nodeID {
+		logrus.Debugf("Volume %v is already attached to node %v", v.Name, v.Spec.NodeID)
+		return v, nil
+	}
+
```

Comment:
Why not add this inside `if v.Spec.NodeID != "" {}`?

Reply:
This is a general idempotency check for volume attachment, by moving it out here.
The if v.Spec.NodeID only needs to deal with migration conditions

Reply:
Seems simpler to me, technically we would have to evaluate all the other params (disableFrontend, lastAttached) as well.
We can add the param validation down the line when we do a refactor of the api/csi driver.
My plan is to make the csi driver dumber and move most of the validation logic into the longhorn api in to the longhorn api/volume manager.